lib/vrl/stdlib/src/parse_key_value.rs

```diff
@@ -822,5 +850,15 @@ mod test {
                              "test": "tom test"})),
             tdef: TypeDef::object(Collection::any()).fallible(),
         }
+
+        duplicate_keys {
+            args: func_args! [
+                value: r#"Cc: "tom" Cc: "bob""#,
+                key_value_delimiter: ":",
+                field_delimiter: " ",
+            ],
+            want: Ok(value!({"Cc": ["tom", "bob"]})),
+            tdef: TypeDef::object(Collection::any()).fallible(),
```

Comment:
Technically I think this type def isn't right, but you aren't making it any worse here. We should be able to specify that the type is an object with string keys and values that are strings, booleans, or arrays of strings.

Reply:
This reminded me that there is the extra case of duplicate keys without values. That shouldn't produce an array.